From 6327c0aa21863e4e960a5862361eefb1c6d3093c Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexdeucher@gmail.com>
Date: Tue, 9 Feb 2010 17:18:48 -0500
Patch-mainline: 2.6.34
References: fate#310916
Git-commit: 12d7798f71107e7f4391a532af8c6690bda16f66
Subject: [PATCH 0418/2588] drm/radeon/kms/evergreen: fix multi-head

Fix a typo in crtc offset setup that set the crtc
instances wrong for all crtcs except the first.

Signed-off-by: Alex Deucher <alexdeucher@gmail.com>
Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit 12d7798f71107e7f4391a532af8c6690bda16f66)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/radeon/atombios_crtc.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/radeon/atombios_crtc.c b/drivers/gpu/drm/radeon/atombios_crtc.c
index 925b453..0ec6934 100644
--- a/drivers/gpu/drm/radeon/atombios_crtc.c
+++ b/drivers/gpu/drm/radeon/atombios_crtc.c
@@ -1145,22 +1145,22 @@ void radeon_atombios_init_crtc(struct drm_device *dev,
 		switch (radeon_crtc->crtc_id) {
 		case 0:
 		default:
-			radeon_crtc->crtc_id = EVERGREEN_CRTC0_REGISTER_OFFSET;
+			radeon_crtc->crtc_offset = EVERGREEN_CRTC0_REGISTER_OFFSET;
 			break;
 		case 1:
-			radeon_crtc->crtc_id = EVERGREEN_CRTC1_REGISTER_OFFSET;
+			radeon_crtc->crtc_offset = EVERGREEN_CRTC1_REGISTER_OFFSET;
 			break;
 		case 2:
-			radeon_crtc->crtc_id = EVERGREEN_CRTC2_REGISTER_OFFSET;
+			radeon_crtc->crtc_offset = EVERGREEN_CRTC2_REGISTER_OFFSET;
 			break;
 		case 3:
-			radeon_crtc->crtc_id = EVERGREEN_CRTC3_REGISTER_OFFSET;
+			radeon_crtc->crtc_offset = EVERGREEN_CRTC3_REGISTER_OFFSET;
 			break;
 		case 4:
-			radeon_crtc->crtc_id = EVERGREEN_CRTC4_REGISTER_OFFSET;
+			radeon_crtc->crtc_offset = EVERGREEN_CRTC4_REGISTER_OFFSET;
 			break;
 		case 5:
-			radeon_crtc->crtc_id = EVERGREEN_CRTC5_REGISTER_OFFSET;
+			radeon_crtc->crtc_offset = EVERGREEN_CRTC5_REGISTER_OFFSET;
 			break;
 		}
 	} else {
-- 
1.7.6

